### ISSUE TITLE
Commandline Tool not returning / returning invalid?

### ISSUE THREAD

**Issue by cluen** [RSuter/NSwag#1226]
Hi,

seems that the commandline tool is returning with an invalid state.
Background: Using nswag command line tool in a postbuild event:
`echo 'Starting nswag'
if "$(BuildingInsideVisualStudio)"=="true" if $(ConfigurationName) == Release  nswag run "$(ProjectDir)\..\$(ProjectName).nswag"
echo 'nswag completed'`

Output from Build:
`1>  'Starting nswag'
1>  NSwag command line tool for .NET 4.6.1+ WinX64, toolchain v11.15.4.0 (NJsonSchema v9.10.29.0 (Newtonsoft.Json v9.0.0.0))
1>  Visit http://NSwag.org for more information.
1>  NSwag bin directory: C:\Program Files (x86)\Rico Suter\NSwagStudio\Win
1>  
1>  Executing file 'C:\TFS\ViMa\AuReS_NEVU_DEV\Source\Reisewitz\Controller\AuReSPlaningController\AuReSPlaningController\\..\AuReSPlaningController.nswag'...
1>  Done.
1>  
1>  Duration: 00:00:04.9409131
`

All commands behind nswag commandline tool are ignored. So ie. the output: "nswag completed" is missing.

**Comment by RSuter** [comment]
Maybe you have to put the echo on a new line? This way it may be passed as arguments to the nswag binary?

**Comment by RSuter** [comment]
I think this is a problem with the new **nswag.cmd** which redirects the call to the correct assembly. 

Can you go to the following path on your system: 
C:\Program Files (x86)\Rico Suter\NSwagStudio

And check the nswag.cmd file? Maybe the problem is inside this and you know how to fix it?

**Comment by RSuter** [comment]
Just to clarify: You are using the NSwag command line tools via the NSwagStudio installer, yes?

**Comment by cluen** [comment]
Yes installed via NSwagStudio Installer

**Comment by cluen** [comment]
Don't know what and how to fix:

This is the actual content:

```
@ECHO OFF
set args=%* 

IF NOT "%args:/runtime:winx86=%" == "%args%" (
    "%~dp0/Win/nswag.x86.exe" %*
    GOTO end
)

IF NOT "%args:/runtime:netcore10=%" == "%args%" (
    dotnet "%~dp0/NetCore10/dotnet-nswag.dll" %*
    GOTO end
)

IF NOT "%args:/runtime:netcore11=%" == "%args%" (
    dotnet "%~dp0/NetCore11/dotnet-nswag.dll" %*
    GOTO end
)

IF NOT "%args:/runtime:netcore20=%" == "%args%" (
    dotnet "%~dp0/NetCore20/dotnet-nswag.dll" %*
    GOTO end
)

"%~dp0/Win/nswag.exe" %*
:end
```

**Comment by cluen** [comment]
Any suggestions? @RSuter ?

**Comment by cluen** [comment]
Any news on this topic?

**Comment by RSuter** [comment]
Have you tried to modify this batch file so that it does not suppress the commands after?

**Comment by cluen** [comment]
No, because I don't know what / in which direction to edit this file.

**Comment by RSuter** [comment]
Try

```
echo 'Starting nswag'
if "$(BuildingInsideVisualStudio)"=="true" if $(ConfigurationName) == Release cmd /c nswag run "$(ProjectDir)\..\$(ProjectName).nswag"
echo 'nswag completed'
```

**Comment by cluen** [comment]
I think I found another solution by invoking nswag in a different way:
https://stackoverflow.com/a/16831269

```
echo 'Starting nswag'
if "$(BuildingInsideVisualStudio)"=="true" if $(ConfigurationName) == Release  nswag run "$(ProjectDir)\..\$(ProjectName).nswag"
echo 'nswag completed'
```

**Comment by RSuter** [comment]
with 

    cmd /c nswag nswag.json

it seems to work
